encoding/protobuf/protoc-gen-yarpc-go/internal/lib/lib.go

```diff
@@ -193,6 +193,81 @@ func Build{{$service.GetName}}YARPCProcedures(server {{$service.GetName}}YARPCSe
 	)
 }
 
+// Provide{{$service.GetName}}YARPCClientParams defines the input
+// for Provide{{$service.GetName}}YARPCClient. It provides the
+// paramaters to get a {{$service.GetName}}YARPCClient in an
+// Fx application.
+type Provide{{$service.GetName}}YARPCClientParams struct {
```

Comment:
nit: Can we get a `$serviceName` variable here given that it's called 4 times in these 5 lines.

Reply:
Typically for Fx modules, we name the Params struct just `Params` or `FooParams`, not `ProvideFooParams`.

Reply:
I can't call this `Params` because there can be multiple services per package. I can make this `FooYARPCClientParams`, but I figured it was more clear to name it `ProvideFooYARPCClientParams`. You're sure I should change this to the former? Preferably, I'd like it to match with whatever we name the function (which shouldn't be Provide as you said).

Reply:
Also, the `$service.GetName` - this is done in 100 other places in this template, I'd want it to be consistent - if it's a big deal, we can change this in a separate PR, but as you said I think it's a nit.

Reply:
Yeah, we can't call it `Params`. `FooYARPCClientParams` is preferable. They're never called `ProvideFooParams` because the constructors are never called `Provide*`.

Reply:
See below comment though - we can't call the functions `New`, so I wanted to relate the structs and functions in some way, esp since the generated code returns `interface{}` from both functions.

Reply:
I mean even if the constructor was called `NewFoo`, the params struct would still be called `FooParams`, not `NewFooParams`.

I think it's easy to say `FooParams` defines the parameters needed to build `Foo` regardless of whether the constructor for it is called `NewFoo` or `ProvideFoo`.